src/Microsoft.DotNet.Tools.Publish/PublishCommand.cs

```diff
@@ -14,124 +14,74 @@
-                }
+                NugetFramework = NuGetFramework.Parse(Framework);
 
-                // Locate the project and get the name and full path
-                var path = project.Value;
-                if (string.IsNullOrEmpty(path))
+                if (NugetFramework.IsUnsupported)
                 {
-                    path = Directory.GetCurrentDirectory();
-                }
-
-                var projectContexts = ProjectContext.CreateContextForEachTarget(path);
-                projectContexts = GetMatchingProjectContexts(projectContexts, nugetframework, runtime.Value());
-
-                if (projectContexts.Count() == 0)
-                {
-                    string errMsg = $"'{project.Value}' cannot be published";
-                    if (framework.HasValue() || runtime.HasValue())
-                    {
-                        errMsg += $" for '{framework.Value()}' '{runtime.Value()}'";
-                    }
-
-                    Reporter.Output.WriteLine(errMsg.Red());
-                    return 1;
-                }
-
-                int result = 0;
-                foreach (var projectContext in projectContexts)
-                {
-                    result += Publish(projectContext, output.Value(), configuration.Value() ?? Constants.DefaultConfiguration);
+                    Reporter.Output.WriteLine($"Unsupported framework {Framework}.".Red());
+                    return false;
                 }
+            }
 
-                return result;
-            });
+            ProjectRidFxPermutations = ProjectContext.CreateContextForEachTarget(ProjectPath);
+            ProjectRidFxPermutations = GetMatchingProjectContexts(ProjectRidFxPermutations, NugetFramework, Runtime);
 
-            try
-            {
-                return app.Execute(args);
-            }
-            catch (Exception ex)
+            if (ProjectRidFxPermutations.Count() == 0)
             {
-#if DEBUG
-                Console.Error.WriteLine(ex);
-#else
-                Console.Error.WriteLine(ex.Message);
-#endif
-                return 1;
+                string errMsg = $"'{ProjectPath}' cannot be published  for '{Framework ?? "<no framework provided>"}' '{Runtime ?? "<no runtime provided>"}'";
+                Reporter.Output.WriteLine(errMsg.Red());
+                return false;
             }
+
+            return true;
         }
 
-        private static bool CheckArg(CommandOption argument)
+        public void PublishAllProjects()
         {
-            if (!argument.HasValue())
+            NumberOfFailedToPublishProjects = 0;
+            foreach (var project in ProjectRidFxPermutations)
             {
-                Reporter.Error.WriteLine($"Missing required argument: {argument.LongName.Red().Bold()}");
-                return false;
+                NumberOfFailedToPublishProjects += PublishRidFxPermutation(project, OutputPath, Configuration ?? Constants.DefaultPublishConfiguration);
```

Comment:
This function is returning an exit code, I don't think it's good to take a dependency on that being equivalent to the number of failed projects